### DISCUSSION TITLE
the dynamic block  name is * U？

### DISCUSSION THREAD

**Discussion by iisaxin** [LibreDWG/libredwg#907] [Q&A]
Excuse me, My program fragments are as follows：
for (i = 0; i < dwg.num_objects; i++)
{
	switch (dwg.object[i].fixedtype)
	{
		case DWG_TYPE_INSERT:
		{
			theInsert=dwg_object_to_INSERT (&dwg.object[i]);
			int error;
			char *insertname=dwg_ref_get_table_name (theInsert->block_header, &error);

General blocks are all given the right names，But if it's a **dynamic block**, then the name is usually *** U**, 
so I want to ask how to get the right result

**Comment by rurban** [comment]
That's the right result. Dynamic blocks, such as parametric blocks, dimension blocks and such always get generated names, starting with a star.

**Reply by iisaxin** [reply]
Thank you for your answer, but this is not the result I want, for example, the actual name is "DIN Rail H5.5", he shows * U, even if he shows "* DIN Rail H5.5", it is ok, the drawing inside this situation is quite much, I hope you can give me a way to solve the problem。
